hubblestack/extmods/comparators/certificate.py

```diff
@@ -0,0 +1,79 @@
+# -*- encoding: utf-8 -*-
+"""
+Certificate comparator used to compare SSL certificate fields
+
+"compare" command example:
+
+    comparator:
+        type: certificate
+        compare:
+            not_before: 30 # maximum number of days until the certificate becomes valid (default value: 0)
+                           # the check is failed if the certificate becomes valid in more than 30 days
+            not_after: 45  # minimum number of days until expiration (default value: 0)
+                           # the check is failed if the certificate expires in less than 45 days
+            fail_if_not_before: True # fails the check if the certificate is not valid yet ( default value: False)
+                           # if True, the check will fail only if not_before is 0 (or missing): if the certificate is not valid yet, but it is expected to be
+                           
+            ssl_issuer_common_name: 'DigiCert SHA2 Secure Server CA', (Optional)
+            ssl_signature_algorithm: 'sha256WithRSAEncryption', (Optional)
+            ssl_subject_country: 'US', (Optional)
+            ssl_subject_organisation: 'Adobe Systems Incorporated', (Optional)
+            ssl_subject_organisation_unit: 'IT' (Optional)
+
+Note: Optional parameters are matched exactly. Currently there is no support for regex match for these params.
+"""
+import logging
+from datetime import datetime
+
+log = logging.getLogger(__name__)
+
+
+def compare(audit_id, result_to_compare, args):
+    """
+    Compare the certificate fields
+    :param audit_id:
+            Check ID
+    :param result_to_compare:
+            Certificate data to compare
+    :param args:
+            Comparator dictionary as mentioned in check
+    """
+    log.debug('Running certificate::compare for check: {0}'.format(audit_id))
+    if result_to_compare.get('ssl_has_expired', False):
```

Comment:
This looks wrong. In comparator, we compare output with user passed values.
Here based on output, we are taking decision. It should be if used has passed this param to compare

Reply:
removing expiry checking code

Reply:
removing this check